lib/ClangImporter/ImportName.cpp

```diff
@@ -1600,6 +1609,14 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
   ArrayRef<const clang::ParmVarDecl *> params;
   switch (D->getDeclName().getNameKind()) {
   case clang::DeclarationName::CXXConstructorName:
+    isInitializer = true;
+    isFunction = true;
+    result.info.initKind = CtorInitializerKind::Designated;
+    baseName = "init";
+    if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(D))
```

Comment:
Re John's earlier comment: it would appear that _on windows only_ `CXXConstructorName` does not necessarily indicate `D` is a `CXXConstructorDecl`. I'll investigate this further because it seems a bit odd.